tools/integration_tests/gzip/gzip_test.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Provides integration tests for gzip objects in gcsfuse mounts.
+package gzip_test
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/helpers"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/mounting/static_mounting"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+const (
+	seqReadSizeMb   = 1
+	textContentSize = 10 * 1e6 * seqReadSizeMb
+
+	textContentWithContentEncodingWithNoTransformFilename    = "textContentWithContentEncodingWithNoTransform.txt"
+	textContentWithContentEncodingWithoutNoTransformFilename = "textContentWithContentEncodingWithoutNoTransform.txt"
+
+	gzipContentWithoutContentEncodingFilename = "gzipContentWithoutContentEncoding.txt.gz"
+
+	gzipContentWithContentEncodingWithNoTransformFilename    = "gzipContentWithContentEncodingWithNoTransform.txt.gz"
+	gzipContentWithContentEncodingWithoutNoTransformFilename = "gzipContentWithContentEncodingWithoutNoTransform.txt.gz"
+
+	gzipTestResPath = "" // setting this non-empty works only if implicit-dirs is passed in mount flags, but that's causing
+	// mount to fail, so temporarily set this to empty.
```

Comment:
@Tulsishah do you know why `implicit-dirs` in flags might cause mount error?

Reply:
I didn't find error while mounting when I pass ```implicit-dirs```  flags. 
What exact error you are facing ?

Reply:
On enabling `--implicit-dirs` I get the following error log:

```
...
2023/08/03 04:04:47 /tmp/gcsfuse_readwrite_test_971925334/bin/gcsfuse --sequential-read-size-mb=1 --implicit-dirs --debug_gcs --debug_fs --debug_fuse --log-file=/tmp/gcsfuse_readwrite_test_971925334/gcsfuse.log --log-format=text gargnitin-gzip-test-bucket /tmp/gcsfuse_readwrite_test_971925334/mnt
2023/08/03 04:04:47 mountGcsfuse: cannot mount gcsfuse: exit status 1
```

On checking `/tmp/gcsfuse_readwrite_test_971925334/gcsfuse.log`, there is nothing in the log:
```
gargnitin@gargnitin:~/work/cloud/storage/client/gcsfuse/src/gcsfuse$ ls -la /tmp/gcsfuse_readwrite_test_971925334/gcsfuse.log
-rw-r--r-- 1 gargnitin primarygroup 278 Aug  3 04:04 /tmp/gcsfuse_readwrite_test_971925334/gcsfuse.log
gargnitin@gargnitin:~/work/cloud/storage/client/gcsfuse/src/gcsfuse$ cat /tmp/gcsfuse_readwrite_test_971925334/gcsfuse.log
/tmp/gcsfuse_readwrite_test_971925334/bin/gcsfuse --sequential-read-size-mb=1 --implicit-dirs --debug_gcs --debug_fs --debug_fuse --log-file=/tmp/gcsfuse_readwrite_test_971925334/gcsfuse.log --log-format=text gargnitin-gzip-test-bucket /tmp/gcsfuse_readwrite_test_971925334/mnt
```

Reply:
@Tulsishah ^

Reply:
Followed up in later PR in https://github.com/GoogleCloudPlatform/gcsfuse/pull/1256/commits/3ac6df6dc005e94f3a2c34964d7d76747bb60f03#r1282722070 .